### ISSUE TITLE
Form name isn't saved when creating a new form

### ISSUE THREAD

**Issue by ziggy-98** [XGovFormBuilder/digital-form-builder#642]
**Describe the bug**
When a new form is created, the name of the new form isn't passed to the newFormJson object which means that although the form has an id, no title is supplied.

**To Reproduce**

1. run `yarn`
2. run `yarn build`
3. click 'Create new form' and give the form a name
4. Once you are passed through to the visualiser, you'll see that the form title box in form details is blank

**Expected behavior**
Form name is passed to the newFormJson object on creation.

**Repository**

- Designer

**Desktop (please complete the following information):**

- OS: MacOS 11.2.1
- Browser: Chrome
- Version: 92

**Proposed solution**
Option 1: Pass through the un-parsed name to the "/api/new" route. Then add the name as a property to the newFormJson object before parsing the name to get the id. 

Options 2: Pass both the parsed name and the un-parsed name in the payload of the route. Then add the un-parsed version as the name to the newFormJson object and pass the parsed name as the id.

If this messes with any coding standards in the project then I'm fine to go away and rethink this approach!

**Comment by jenbutongit** [comment]
hey @ziggy-98, thanks for looking into this.. sorry for the daft question, what's the issue you're facing?

**Comment by ziggy-98** [comment]
Hey @jenbutongit - sorry looking at it now my description was a bti jumbled! Basically the issue is that when a new form is created the upload function is passed a default form object without a name property. This means that the form title field in the form details section is blank and if you try to edit it and save the new form name it breaks the designer.

TLDR; when a form is created the title isn't added to the form.

Hope that makes sense! Like I said I can make a quick change to pass through the name separately from the id in the save function so that it adds both to the form before uploading, then submit a PR if you want 😄

**Comment by jenbutongit** [comment]
ok, I think I understand now. I think title was introduced as part of feedback forms (however I'd rather tear that out and properly design this feature, if we need it at all..!). In any case, ID !== title. 

What are you trying to achieve? being able to change the title/id of the form within the designer? 

I actually know what the issue is with the title.. https://github.com/XGovFormBuilder/digital-form-builder/blob/main/model/src/schema/schema.ts. Since title isn't an expected field, joi is validating based on this schema, and then throwing an error (so it doesn't save in the designer).  

for context, when forms are loaded up in the runner (see `runner/src/server/forms`), we use the file name, so `test.json` will be loaded at `localhost:3009/test`

**Comment by ziggy-98** [comment]
Ah ok I see now, so is the service name at the top of the screen just formatted from the id of the form? I guess I was getting confused because I noticed that there was a key called "name" on the form schema but I've noticed now that field is optional so I guess it's not a huge issue.

From now on I'll just use the id for the title and format it differently. Like you say maybe the field isn't even needed, thanks for you help!
